src/test/java/redis/clients/jedis/authentication/TokenBasedAuthenticationClusterIntegrationTests.java

```diff
@@ -0,0 +1,131 @@
+package redis.clients.jedis.authentication;
+
+import static org.hamcrest.Matchers.greaterThanOrEqualTo;
+import static org.junit.Assert.assertEquals;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.atLeast;
+import static org.mockito.Mockito.doAnswer;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.verify;
+import static org.awaitility.Awaitility.await;
+import static org.awaitility.Durations.*;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+
+import org.junit.Test;
+
+import redis.clients.authentication.core.IdentityProvider;
+import redis.clients.authentication.core.IdentityProviderConfig;
+import redis.clients.authentication.core.SimpleToken;
+import redis.clients.authentication.core.Token;
+import redis.clients.authentication.entraid.EntraIDTokenAuthConfigBuilder;
+import redis.clients.jedis.Connection;
+import redis.clients.jedis.ConnectionPoolConfig;
+import redis.clients.jedis.DefaultJedisClientConfig;
+import redis.clients.jedis.HostAndPort;
+import redis.clients.jedis.HostAndPorts;
+import redis.clients.jedis.JedisClientConfig;
+import redis.clients.jedis.JedisCluster;
+import redis.clients.jedis.JedisClusterTestBase;
+
+public class TokenBasedAuthenticationClusterIntegrationTests extends JedisClusterTestBase {
+
+    @Test
+    public void testClusterInitWithAuthXManager() {
+        IdentityProviderConfig idpConfig = new IdentityProviderConfig() {
+            @Override
+            public IdentityProvider getProvider() {
+                return new IdentityProvider() {
+                    @Override
+                    public Token requestToken() {
+                        return new SimpleToken("default", "cluster",
+                                System.currentTimeMillis() + 5 * 1000, System.currentTimeMillis(),
+                                null);
+                    }
+                };
+            }
+        };
+        AuthXManager manager = new AuthXManager(EntraIDTokenAuthConfigBuilder.builder()
+                .lowerRefreshBoundMillis(1000).identityProviderConfig(idpConfig).build());
+
+        HostAndPort hp = HostAndPorts.getClusterServers().get(0);
```

Comment:
You should use here `cluster-entraid-acl` endpoint

Reply:
as long as this endpoint answers to cluster commands it would be fine. Right now the tests can run without  entraid enabled endpoints and make the necessary assertions on it. It would be better to keep them running on each commit.

Reply:
In the integration test, we do not bring up the endpoints hardcoded in the cluster config, so we need to switch to the `cluster-entraid-acl` to be able to run the tests. Let me know if I'm missing something here.

Reply:
yes , it doesnt run with entraid enabled env but runs within every integration test in jedis repo